### DISCUSSION TITLE
Integrating multiple external html files in randomized order into jspsych

### DISCUSSION THREAD

**Discussion by vscholz118** [jspsych/jsPsych#1338] [Q&A]
Hello everyone,
I really hope someone here may be able to help me. I have html files for some questionnaire and before serving them up to participants, I will need to randomize their order and then serve them up in the randomized order before saving the data. The html files are programmed so that each time participants are finished with one, they click a button to get to the next page. This is the code I have so far but I got stuck and I have no idea how to continue e.g. would I need a timeline for serving up the different html files or does this not work for external html files? Maybe I need a completely different approach here?
I'm really hoping someone has some advise for me and I'm sorry if this question is very basic. I have been searching for a solution online but so far I have not found a direct solution for my problem.
Thank you very much!

````
// get  questionnaires, randomize their order of presentation,
// serve them up in this order and save data in csv

var quest1 = {
    type:'external-html',
    url: "quest1.html",
    cont_btn: "start",
}

var quest2 = {
    type:'external-html',
    url: "quest2.html",
    cont_btn: "start",
}

var quest3 = {
    type:'external-html',
    url: "quest3.html",
    cont_btn: "start",
}

var myArray = [ quest1, quest2, quest3];
var Quest_Shuff = jsPsych.randomization.shuffle(myArray);

var ClinQuest = {   // how to run all html files after another in Quest_Shuff order??

                           on_finish: function(data) {
                                   var quest_data = jsPsych.data.getTrialsOfType('external-html'); // or other trial type?
                                   saveData("filename.csv", jsPsych.data.dataAsCSV()); 
                         }
````

**Comment by jodeleeuw** [comment]
Hi @vscholz118,

If your html files are compatible with the `external-html` plugin, then what you've got is pretty close to what you need to do. You can use your `Quest_Shuff` variable and make it the timeline.

```js
jsPsych.init({
	timeline: Quest_Shuff
})
```

Or, if you have other trials as part of the experiment you can add the `Quest_Shuff` trials like this:

```js
var timeline = [];

// other stuff added to timeline

timeline = timeline.concat(Quest_Shuff)

jsPsych.init({
	timeline: timeline
});
```
